### PR TITLE
GUI Improvement

### PR DESCRIPTION
**Pull request by mel-odious**
Some of the changes discussed in [this thread](http://www.freeorion.org/forum/viewtopic.php?f=9&t=9470).
Main changes are:
- Add size/position validation to CUIWnd::SizeMove() (removed similar code from LDrag()),
- Override CUIWnd::Run() to register modal CUIWnds with the HumanClientApp instance so that they can have their positions validated when the app window size changes,
- Change MapWnd::DoLayout() to validate positions of the various windows available in the map view (sitreps, combat report etc.),
- Refactor InGameMenu to recalculate position in InGameMenu::SizeMove() (this keeps it centered when the app is resized)

### CODE REVIEW

File: UI/CUIWnd.cpp [hunk 1]
```diff
@@ -153,13 +154,41 @@ CUIWnd::CUIWnd(const std::string& t, GG::X x, GG::Y y, GG::X w, GG::Y h, GG::Fla
     SetMinSize(GG::Pt(CUIWnd::MinimizedWidth(), BORDER_TOP + INNER_BORDER_ANGLE_OFFSET + BORDER_BOTTOM + 50));
     InitButtons();
     SetChildClippingMode(ClipToClientAndWindowSeparately);
+    ValidatePosition();
 }
 
 CUIWnd::~CUIWnd()
-{}
+{ }
```
Comment (geoffthemedio) [inline UI/CUIWnd.cpp]: Please leave (and add) {} not { }

**Comment by geoffthemedio** [comment]
Are you planning to come back to this any time soon?

**Comment by mel-odious** [comment]
Definitely, I've just been playing the game for a bit instead of developing :)
Should be able to get back to it in a couple of days.

(ETA: clicked the 'close and comment' button by mistake)

File: GG/GG/GUI.h [hunk 2]
```diff
@@ -6,12 +6,12 @@
    modify it under the terms of the GNU Lesser General Public License
    as published by the Free Software Foundation; either version 2.1
    of the License, or (at your option) any later version.
-
```
Comment (geoffthemedio) [inline GG/GG/GUI.h:9]: Why are you editing the copyright notice whitespace?

**Comment by geoffthemedio** [comment]
Can't auto merge; can you fix conflicts?

File: GG/GG/GUI.h (hunk 2 above)
Comment (mel-odious) [inline GG/GG/GUI.h:9]: I have an extension that automatically removes trailing whitespace.  I can disable it and restore the whitespace if this is an issue.

Comment (geoffthemedio) [inline GG/GG/GUI.h:9]: In the midst of a bunch of unrelated changes, it's distracting and arbitrary seeming.

Comment (mel-odious) [inline GG/GG/GUI.h:9]: OK, I'll disable it in future.

**Comment by mel-odious** [comment]
This branch is pretty old.  I'll try rebasing (cleaning up the commits in the process...) and then I could force push to my fork (which would update this PR automatically) as suggested in [this article](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) or just open a new PR.

**Comment by geoffthemedio** [comment]
Sounds find to me... I don't know why anyone would have sub-branched off your branch, or why you should need to worry about what happens if they did, which seems to have been the main objection to doing that.

**Comment by mel-odious** [comment]
(not ready for pull yet, I'm finding bugs to squash after the rebase)

**Comment by mel-odious** [comment]
I think I found everything.  I noticed that MapWnd::m_moderator_wnd seems to be allocated and "owned" by MapWnd, but I couldn't find where it's deleted.  Is it supposed to outlive MapWnd or is it just a leak?

**Comment by geoffthemedio** [comment]
I don't recall what I was planning when adding it, but it's possible it's a nominal leak... but which doesn't matter since I suspect the only time MapWnd is deleted is when closing the program. Likely this is a separate cleanup/bugfix issue to this set of changes, though.
